xapian-applications/omega/query.cc

```diff
@@ -1940,6 +1955,19 @@ eval(const string &fmt, const vector<string> &param)
 		value = args[0];
 		url_prettify(value);
 		break;
+	    case CMD_qid: {
+		string qstr = args[0];
+		trim_string(qstr);
```

Comment:
What is using this hashed query id supposed to accomplish?

If it's to track a particular user's "search session", then trimming spaces doesn't seem necessary - users don't commonly repeat the exact same query but add or remove spaces at the start or end or it.

If we're trying to normalise query strings which are really the same, then it doesn't really seem to be doing enough.  E.g. something like `$transform{ $query ,\s+, ,g}` would normalise all runs of whitespace to a single space (and put a single space at the start and end - that's just because it's simpler to ensure a single space there than none).  Or perhaps we should go further and use the terms parsed from the query or something, so that punctuation and word order (outside of phrases) don't matter.